_drafts/javascript.md

```diff
@@ -0,0 +1,463 @@
+
+   ```
+   npm install -g yo generator-hubot
+   mkdir my-awesome-hubot && cd my-awesome-hubot
+   yo hubot --adapter=slack
+   npm link hubot
+   ```
+
+3. Now start hubot
+
+   ```
+   NODE_PATH=`pwd`/node_modules HUBOT_SLACK_TOKEN=<your token here> ./bin/hubot --adapter slack
+   ```
+
+   replace `<your token here>` with your own bots token from slack
+
+### Convert test files from CoffeeScript to JavaScript
+
+Same as the previous steps only for the tests files this time.
+
+When finished, create a separate branch with the tests written in JavaScript but the source code still in CoffeeScript to assure integrity.
+
+### Update package.json
+
+Remove all CoffeeScript related dependencies and tooling. Update [engines](https://docs.npmjs.com/files/package.json#engines)
+
+Make sure that Hubot scripts can still be written in CoffeeScript. We want to break as few existing scripts as possible and many of them are written in CoffeeScript. We treat it as legacy support and will drop support for scripts written in CoffeeScript in future.
+
+### Add script for JavaScript linting
+
+The only tool I would like to introduce as part of this proposal is a linting tool which will be run as part of `npm test`.
+
+I suggest [standard](https://standardjs.com/). We’ve been using it in all our projects at Hoodie and Neighbourhoodie since 2+ years and never looked back. It's a zero-configuration JavaScript linter.
+
+### Release `hubot@3.0.0`
+
+#### Update hubot’s version range `peerDependencies` of default scripts
+
+- [ ] [hubot-scripts/hubot-diagnostics](https://github.com/hubot-scripts/hubot-diagnostics)
+- [ ] [hubot-scripts/hubot-help](https://github.com/hubot-scripts/hubot-help)
+- [ ] [hubot-scripts/hubot-heroku-keepalive](https://github.com/hubot-scripts/hubot-heroku-keepalive)
+- [ ] [hubot-scripts/hubot-google-images](https://github.com/hubot-scripts/hubot-google-images)
+- [ ] [hubot-scripts/hubot-google-translate](https://github.com/hubot-scripts/hubot-google-translate)
+- [ ] [hubot-scripts/hubot-pugme](https://github.com/hubot-scripts/hubot-pugme)
+- [ ] [hubot-scripts/hubot-redis-brain](https://github.com/hubot-scripts/hubot-redis-brain)
+- [ ] [hubot-scripts/hubot-rules](https://github.com/hubot-scripts/hubot-rules)
+- [ ] [hubot-scripts/hubot-shipit](https://github.com/hubot-scripts/hubot-shipit)
+
+replace
+
+    "peerDependencies": {
+      "hubot": "2.x"
+    },
+
+with
+
+    "peerDependencies": {
+      "hubot": ">=2 <10"
+    },
+
+That will give us enough space to follow semantic-versioning with `hubot` while not breaking the eco system each time.
+
+Once we decide that it’s time to make a breaking change which breaks existing adapters, we jump version numbers and go directly to `hubot@10`. Semantic versioning does not forbid that.
+
+#### Publish `hubot@3.0.0` with `--tag=next`
+
+- Setup semantic-versioning for `hubot` as described in [`semantic-release` proposal](https://github.com/hubotio/evolution/pull/6).
```

Comment:
Sorry for the confusion, I changed my mind here, I think we can use semantic-release for version 3.0.0 already as we will release with the `@next` tag at first